java/clients/src/main/java/sleeper/clients/deploy/jar/PublishJarsToRepo.java

```diff
@@ -0,0 +1,115 @@
+/*
+ * Copyright 2022-2025 Crown Copyright
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package sleeper.clients.deploy.jar;
+
+import sleeper.clients.util.command.CommandPipelineRunner;
+import sleeper.clients.util.command.CommandUtils;
+import sleeper.core.SleeperVersion;
+import sleeper.core.deploy.ClientJar;
+import sleeper.core.deploy.LambdaJar;
+
+import java.io.IOException;
+import java.nio.file.Path;
+
+import static java.util.Objects.requireNonNull;
+
+public class PublishJarsToRepo {
+    private final Path filePath;
+    private final String repoUrl;
+    private final String version;
+    private final CommandPipelineRunner commandRunner;
+
+    private PublishJarsToRepo(Builder builder) {
+        this.filePath = requireNonNull(builder.filePath, "File path must not be null");
+        this.repoUrl = requireNonNull(builder.repoUrl, "Repository URL must not be null");
+        this.version = requireNonNull(builder.version, "Version to publish must not be null");
+        this.commandRunner = requireNonNull(builder.commandRunner, "Command Runner must not be null");
+    }
+
+    public static Builder builder() {
+        return new Builder();
+    }
+
+    public static void main(String[] args) throws Exception {
+        builder()
+                .filePath(Path.of(args[0] + "/"))
+                .repoUrl(args[1])
+                .build()
+                .upload();
+
+    }
+
+    public void upload() {
+        for (ClientJar clientJar : ClientJar.getAll()) {
+            deployJars(clientJar.getFilenameFormat(), clientJar.getArtifactId());
+        }
+
+        for (LambdaJar lambdaJar : LambdaJar.getAll()) {
+            deployJars(lambdaJar.getFilenameFormat(), lambdaJar.getArtifactId());
+        }
+    }
+
+    //Requires matching server with auth details in local m2 settings.xml <servers>
+    private void deployJars(String filename, String imageName) {
+        try {
+            commandRunner.run("mvn", "deploy:deploy-file", "-q",
+                    "-Durl=" + repoUrl,
+                    "-DrepositoryId=repo.id", //Requires matching server with auth details in local m2 settings.xml <servers>
+                    "-Dfile=" + filePath.resolve(String.format(filename, version)),
```

Comment:
We're relying on this String.format call being the same as the logic in ClientJar and LambdaJar. We can just ask those classes to do it and have it in one place.

The parameter name also seems incorrect at the moment, but it would be correct if we pass in the result from ClientJar or LambdaJar.